CHANGES.rst

```diff
@@ -4,7 +4,8 @@ Changes
 4.4.3 (unreleased)
 ------------------
 
-- Nothing changed yet.
+- Add support for general-purpose Python 3.x annotations / type-hints in interface definitions.
```

Comment:
Does mypy think that `adapter = IAdapter(context)` produces an *instance* of `IAdapter`, and hence tries to propagate type information from `IAdapter`? If so, I can't decide if that's clever or silly of it, especially since there's no `__init__` method defined...

If `name: str = Attribute('name')` produces errors in mypy, have you considered using the external files support (.pyi)? Does that have the same problem? (I guess you could use a union type Union[str,Attribute] if so.)

Note that I'm not trying to say I've changed my mind and consider this a bad idea, I'm just trying to (a) understand your use case and try to help solve it better and (b) make sure we don't set expectations for `__annotations__` that aren't actually met---perhaps a more clear change note would just explicitly say something about attributes and mypy.